src/pages/index.js

```diff
@@ -97,10 +102,125 @@ const newCardPopup = new PopupWithForm(
 );
 newCardPopup.setEventListeners();
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+//Profile creation and editing
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+api
+  .getUser()
+  .then((inputValues) => {
+    userInfo.setUserInfo(inputValues.name, inputValues.about);
+    userInfo.setAvatar(inputValues.avatar);
+  })
+  .catch((err) => {
+    console.error("Error getting user information:", err);
+  });
+
+function handleProfileEditSubmit(inputValues) {
+  profileEditPopup.renderLoading(true);
+  api
+    .editProfile(inputValues.name, inputValues.about)
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(() => {
+      profileEditPopup.renderLoading(false);
+      profileEditPopup.close();
+    });
+}
+
+profileEditButton.addEventListener("click", () => {
+  const data = userInfo.getUserInfo();
+  profileTitleInput.value = data.name;
+  profileDescriptionInput.value = data.about;
+  profileEditPopup.open();
+});
+
+//DELETING CARDS
 
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
 profileEditPopup.setEventListeners();
+
+const deleteCardPopup = new PopupConfirmDelete("#delete-card-modal");
+
+function confirmDelete(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setEventListeners();
```

Comment:
`deleteCardPopup.setEventListeners();` should be in line 155 

* There are errors when I delete 3 cards in a row https://snipboard.io/st571v.jpg 


`.setEventListeners()` should be removed from `opening` because it's a **public** method that should be called only **once** in `index.js` otherwise you add listeners again and again to the same elements. This can cause a **`memory leak`**

Reply:
That fixed it! Thank you.